sycl/include/sycl/handler.hpp

```diff
@@ -1019,6 +927,9 @@ class __SYCL_EXPORT handler {
 
   bool eventNeeded() const;
 
+  device get_device() const;
```

Comment:
That's an extra `std::shared_ptr` copy. https://github.com/intel/llvm/pull/20698/files#r2561353949 is related. Can you collaborate with @lbushi25 for a better fix?